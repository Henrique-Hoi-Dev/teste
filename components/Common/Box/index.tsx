import React from "react";
import styles from "./styles.module.scss";

interface Props {
  children: React.ReactNode;
  className?: string;
}

export function Box({ className = "", children }: Props) {
  return (
    <div className={`${styles["box-container"]} ${className}`}>{children}</div>
  );
}

export default Box;
