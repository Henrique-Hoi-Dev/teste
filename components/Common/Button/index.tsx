import React from "react";
import styles from "./styles.module.scss";

interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  buttonType: "primary" | "secondary" | "terciary";
  disabledPartner?: boolean;
  beAnAdvisorForm?: boolean;
  icon?: React.ReactNode;
  disabled?: boolean;
}
export function Button({
  children,
  onClick,
  buttonType,
  beAnAdvisorForm,
  disabledPartner,
  icon,
  disabled,
  ...rest
}: ButtonProps) {
  const buttonTypesClass = {
    primary: "button--primary",
    secondary: "button--secondary",
    terciary: "button--terciary",
  };

  const iconButtonClassName = icon ? styles["button--icon"] : "";

  return (
    <button
      disabled={disabled}
      className={`${styles.button} ${
        styles[buttonTypesClass[buttonType]]
      } button ${iconButtonClassName} ${
        !!disabledPartner && styles["button--disabled-home"]
      } ${!!beAnAdvisorForm && styles["button--be-an-advisor"]}`}
      onClick={onClick}
      {...rest}
    >
      {children} {icon}
    </button>
  );
}
