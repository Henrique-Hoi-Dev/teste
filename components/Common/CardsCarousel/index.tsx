import React, { useRef, useState } from "react";
import styles from "./styles.module.scss";
import "slick-carousel/slick/slick.css";
import "slick-carousel/slick/slick-theme.css";
import Slider, { Settings } from "react-slick";
import { MdOutlineNavigateNext, MdOutlineNavigateBefore } from "react-icons/md";
import { GetScreenType } from "@/utils/hooks/getScreenType";
import { TABLET_BREAKPOINT } from "@/utils/breakpoints";

interface Props {
  children: React.ReactNode;
  itemsLength: number;
  className?: string;
  customSlidesPerBreakpoint?: {
    desktop: number;
    tablet: number;
    mobile: number;
  };
  orangeArrows?: boolean;
  arrowsContainerClassName?: string;
  customBreakpoint?: CustomBreakpoint;
  customPixelsBreakpoint?: Omit<CustomBreakpoint, "desktop">;
  arrowsPosition?: "top" | "bottom";
  header?: React.ReactNode;
  onSlideChange?: (index: number) => void;
}

interface CustomBreakpoint {
  desktop: number;
  tablet: number;
  mobile: number;
}

export function CardsCarousel(props: Props) {
  const {
    children,
    className,
    itemsLength,
    orangeArrows,
    arrowsContainerClassName = "",
    customPixelsBreakpoint,
    customBreakpoint,
    arrowsPosition = "bottom",
    header,
    onSlideChange,
  } = props;
  const arrowButtonStylization = orangeArrows
    ? {
        style: {
          backgroundColor: "#F58220",
          color: "white",
        },
      }
    : {};

  const [currentSlide, setCurrentSlide] = useState(0);

  const defaultBreakpoints = {
    desktop: 2,
    tablet: 2,
    mobile: 1,
  };
  const defaultScreenBreakpoints = {
    mobile: 585,
    tablet: TABLET_BREAKPOINT,
  };

  const customScreenBreakpoints =
    customPixelsBreakpoint || defaultScreenBreakpoints;

  const screen = GetScreenType(customScreenBreakpoints);

  const handleArrowsDisplay = () => {
    if (screen === "mobile") return itemsLength > 1;

    return itemsLength > 2;
  };

  const shouldRenderArrows = handleArrowsDisplay();

  const slidesPerBreakpoint = customBreakpoint || defaultBreakpoints;

  const slidesToScrollPerBreakpoint = {
    desktop: 1,
    tablet: 1,
    mobile: 1,
  };

  const settings: Settings = {
    dots: false,
    speed: 500,
    infinite: false,
    slidesToShow: slidesPerBreakpoint["desktop"],
    slidesToScroll: slidesToScrollPerBreakpoint["desktop"],
    afterChange: (currentSlide) => setCurrentSlide(currentSlide),
    arrows: false,
    responsive: [
      {
        breakpoint: customScreenBreakpoints["mobile"],
        settings: {
          slidesToShow: slidesPerBreakpoint["mobile"],
          slidesToScroll: slidesPerBreakpoint["mobile"],
        },
      },
      {
        breakpoint: customScreenBreakpoints["tablet"],
        settings: {
          slidesToShow: slidesPerBreakpoint["tablet"],
          slidesToScroll: slidesToScrollPerBreakpoint["tablet"],
        },
      },
    ],
  };

  const sliderRef = useRef<Slider>(null);

  const goToNextSlide = () => {
    if (sliderRef.current) sliderRef.current.slickNext();
    setCurrentSlide((currentSlide) => currentSlide + 1);

    onSlideChange && onSlideChange(currentSlide + 1);
  };

  const goToPreviousSlide = () => {
    if (sliderRef.current) sliderRef.current.slickPrev();
    setCurrentSlide((currentSlide) => currentSlide - 1);

    onSlideChange && onSlideChange(currentSlide - 1);
  };

  const isPreviousArrowDisabled = currentSlide === 0;
  const isNextArrowDisabled =
    currentSlide + 1 > itemsLength - slidesPerBreakpoint[screen];

  const renderCustomArrows = () => (
    <div className={`${styles["custom-arrows"]} ${arrowsContainerClassName}`}>
      <button
        {...arrowButtonStylization}
        onClick={goToPreviousSlide}
        disabled={isPreviousArrowDisabled}
      >
        <MdOutlineNavigateBefore />
      </button>
      {header ? header : ""}
      <button
        {...arrowButtonStylization}
        onClick={goToNextSlide}
        disabled={isNextArrowDisabled}
      >
        <MdOutlineNavigateNext />
      </button>
    </div>
  );

  return (
    <section className={styles["cards-carousel"]}>
      <div>
        {arrowsPosition === "top" && shouldRenderArrows && renderCustomArrows()}
      </div>
      <Slider
        className={`${styles["cards-carousel__slider"]} ${className}`}
        {...settings}
        ref={sliderRef}
      >
        {children}
      </Slider>
      {arrowsPosition === "bottom" &&
        shouldRenderArrows &&
        renderCustomArrows()}
    </section>
  );
}
