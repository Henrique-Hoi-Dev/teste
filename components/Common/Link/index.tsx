import React, { MouseEvent, useCallback } from "react";
import NextLink, { LinkProps as NextLinkProps } from "next/link";
import styles from "./styles.module.scss";

interface LinkProps extends NextLinkProps {
  children: React.ReactNode;
  variant?: "outlined" | "contained";
  className?: string;
  target?: "_blank" | "_self";
  href?: string;
}

export function Link({
  children,
  variant,
  className,
  href,
  target,
  ...rest
}: LinkProps) {
  const linkCustomClassNames = {
    contained: "link__contained",
    outlined: "link__outlined"
  };

  let customClassName = variant ? linkCustomClassNames[variant] : "";

  const handleClick = useCallback(
    (e: MouseEvent<HTMLAnchorElement>) => {
      if ((href as string).startsWith("#")) {
        e.preventDefault();
        const destination = document.querySelector(href as string);
        if (destination) {
          destination.scrollIntoView({
            behavior: "smooth"
          });
        }
      }
    },
    [href]
  );

  return (
    <NextLink href={href} {...rest}>
      <a
        className={`${styles.link} ${styles[customClassName]} ${className}`}
        target={target || "_self"}
        onClick={handleClick}
      >
        {children}
      </a>
    </NextLink>
  );
}
