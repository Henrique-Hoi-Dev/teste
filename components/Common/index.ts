export * from "./Accordion";
export * from "./Box";
export * from "./Button";
export * from "./CardCarousel";
export * from "./CardsCarousel";
export * from "./CardVideo";
export * from "./LineChart";
export * from "./Table";
export * from "./VideosSection";
export * from "../Banner";
export * from "./FrequentlyAskedQuestions";
export * from "./SmallBanner";
export * from "./CardWithBackground";
export * from "./Link";
export * from "./LinkedinShareButton";
export * from "./StrapiResponsiveImage";
export * from "./Breadcrumb";
