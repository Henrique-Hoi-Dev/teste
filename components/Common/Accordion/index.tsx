/* eslint-disable @next/next/no-img-element */
import { Button } from "@/components/Common/Button";
import { ChevronDown } from "@/icons/ChevronDown";
import { ChevronUp } from "@/icons/ChevronUp";
import { useState } from "react";
import styles from "./styles.module.scss";

interface AccordionProps {
  title: string;
  subtitle?: string;
  description: string;
  shownDescriptionText?: string;
  hiddenDescriptionText?: string;
}

export function Accordion({
  title,
  subtitle,
  description,
  shownDescriptionText = "Esconder mensagem",
  hiddenDescriptionText = "Ver mensagem",
}: AccordionProps) {
  const [shouldShowDescription, setShouldShowDescription] = useState(false);

  function handleToggleShouldShowSpeechDescription() {
    setShouldShowDescription((oldState) => !oldState);
  }

  return (
    <section
      aria-labelledby="accordion"
      className={styles["accordion-container"]}
    >
      <h3 id="accordion">{title}</h3>
      {subtitle && <p>{subtitle}</p>}

      <Button
        data-testid="accordion-button"
        buttonType="primary"
        onClick={handleToggleShouldShowSpeechDescription}
      >
        <div className={styles["accordion-button"]}>
          {shouldShowDescription ? (
            <>
              <span>{shownDescriptionText}</span>
              <ChevronUp />
            </>
          ) : (
            <>
              <span>{hiddenDescriptionText}</span>
              <ChevronDown />
            </>
          )}
        </div>
      </Button>
      {shouldShowDescription && <p>{description}</p>}
    </section>
  );
}
