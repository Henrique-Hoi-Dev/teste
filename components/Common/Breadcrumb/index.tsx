import React from "react";
import Link from "next/link";
import { BsChevronLeft } from "react-icons/bs";
import styles from "./styles.module.scss";

interface Props {
  currentRouteTitle: string;
  previousRoute: {
    label: string;
    href: string;
  };
  className?: string;
}

export function BreadCrumb({
  currentRouteTitle,
  previousRoute,
  className = "",
}: Props) {
  return (
    <header className={`${styles.breadcrumb}  ${className}`}>
      <div className="container">
        <Link href={previousRoute.href}>
          <a>
            <BsChevronLeft /> {previousRoute.label}{" "}
          </a>
        </Link>
        <span> | </span>
        <strong> {currentRouteTitle} </strong>
      </div>
    </header>
  );
}
