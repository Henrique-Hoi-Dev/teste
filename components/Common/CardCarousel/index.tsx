import React, { useRef, useState } from "react";
import "slick-carousel/slick/slick.css";
import "slick-carousel/slick/slick-theme.css";
import Slider, { Settings } from "react-slick";
import { CardCarousel as CardCarouselProps } from "@/interfaces/CardCarousel";
import { CardWithBackground } from "../CardWithBackground";
import { Button } from "@/components/Common/Button";
import { MOBILE_BREAKPOINT, TABLET_BREAKPOINT } from "@/utils/breakpoints";
import { MdOutlineNavigateNext, MdOutlineNavigateBefore } from "react-icons/md";
import { GetScreenType } from "@/utils/hooks/getScreenType";
import styles from "./styles.module.scss";

interface Props extends CardCarouselProps {
  wrapperClassName?: string;
  cardClassName?: string;
  slidesPerBreakpoint?: {
    desktop: number;
    mobile: number;
    tablet: number;
  };
}

export function CardCarousel(props: Props) {
  const {
    title,
    cards,
    isBeAnAdvisorCard,
    wrapperClassName = "",
    cardClassName = "",
    slidesPerBreakpoint,
  } = props;

  const sliderRef = useRef<Slider>(null);
  const [currentSlide, setCurrentSlide] = useState(0);
  const screen = GetScreenType();

  const isNextButtonDisabled = () => {
    return screen === "desktop"
      ? currentSlide + slidesToScroll[screen] >=
          Math.floor(cards.length / slidesToScroll[screen]) - 1
      : Math.ceil(currentSlide + 1) >= cards.length;
  };

  const defaultSlidesPerBreakpointConfig = {
    desktop: 2,
    mobile: 1,
    tablet: 1.3,
  };

  const slidesToScroll = {
    desktop: 1,
    mobile: 1,
    tablet: 1,
  };

  const slidesToShow = slidesPerBreakpoint
    ? slidesPerBreakpoint
    : defaultSlidesPerBreakpointConfig;

  const settings: Settings = {
    dots: false,
    speed: 500,
    infinite: false,
    slidesToShow: slidesToShow["desktop"],
    slidesToScroll: slidesToScroll["desktop"],
    arrows: false,
    beforeChange: (_oldIndex, newIndex) => {
      setCurrentSlide(newIndex);
    },
    responsive: [
      {
        breakpoint: MOBILE_BREAKPOINT,
        settings: {
          slidesToShow: slidesToShow["mobile"],
          slidesToScroll: slidesToScroll["mobile"],
        },
      },
      {
        breakpoint: TABLET_BREAKPOINT,
        settings: {
          slidesToShow: slidesToShow["tablet"],
          slidesToScroll: slidesToScroll["tablet"],
        },
      },
    ],
  };

  const goToNextSlide = () => {
    if (sliderRef.current) sliderRef.current.slickNext();
    setCurrentSlide((currentSlide) => currentSlide + 1);
  };

  const goToPreviousSlide = () => {
    if (sliderRef.current) sliderRef.current.slickPrev();
    setCurrentSlide((currentSlide) => currentSlide - 1);
  };

  const renderCustomArrows = () => (
    <div className={styles["custom-arrows"]}>
      <button onClick={goToPreviousSlide}>
        <MdOutlineNavigateBefore />
      </button>
      <button onClick={goToNextSlide}>
        <MdOutlineNavigateNext />
      </button>
    </div>
  );

  const renderCustomArrowsAdvisorPage = () => {
    return (
      <div className={styles["custom-arrows"]}>
        <Button
          buttonType="primary"
          disabled={currentSlide === 0}
          onClick={goToPreviousSlide}
        >
          <div className={styles["button-arrow-container"]}>
            <MdOutlineNavigateBefore />
          </div>
        </Button>
        <Button
          buttonType="primary"
          disabled={isNextButtonDisabled()}
          onClick={goToNextSlide}
        >
          <div className={styles["button-arrow-container"]}>
            <MdOutlineNavigateNext />
          </div>
        </Button>
      </div>
    );
  };

  const advisorPageArrows = isBeAnAdvisorCard
    ? renderCustomArrowsAdvisorPage()
    : renderCustomArrows();

  const isArrowVisible =
    screen === "mobile" ? cards.length > 1 : cards.length > 2;

  const renderSlider = () => (
    <Slider
      className={`${styles["card-carousel__cards-slider"]} ${wrapperClassName}`}
      {...settings}
      ref={sliderRef}
    >
      {renderCards()}
    </Slider>
  );

  const renderCards = () =>
    cards.map((card) => (
      <div
        className={`${styles["card-carousel__card"]} ${cardClassName}`}
        key={card.id}
      >
        <CardWithBackground
          {...card}
          key={card.id}
          isBeAnAdvisorCard={isBeAnAdvisorCard}
        />
      </div>
    ));

  return (
    <section className={styles["card-carousel"]}>
      <strong className={styles["card-carousel__title"]}>
        {isBeAnAdvisorCard ? title.toUpperCase() : title.toUpperCase()}
      </strong>
      {renderSlider()}
      {isArrowVisible && advisorPageArrows}
    </section>
  );
}
