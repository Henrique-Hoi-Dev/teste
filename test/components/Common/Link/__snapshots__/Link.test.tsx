import { Link } from "@/components/Common/Link";
import { render, screen } from "@testing-library/react";

describe("Link", () => {
  it("Link renders, and have the right href", () => {
    render(<Link href="/home"> Home Link </Link>);

    const link = screen.getByRole("link", { name: "Home Link" });
    expect(link).toHaveProperty("href", expect.stringContaining("/home"));
  });

  it("renders link unchanged", () => {
    const { container } = render(<Link href="/home">Link</Link>);
    expect(container).toMatchSnapshot();
  });
});
