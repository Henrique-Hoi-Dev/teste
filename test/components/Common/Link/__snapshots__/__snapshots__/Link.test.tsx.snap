// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders link unchanged 1`] = `
<div>
  <a
    class="link  undefined"
    href="/home"
  >
    Link
  </a>
</div>
`;
